src/map/clif.c

```diff
@@ -20226,6 +20226,187 @@ void clif_skill_scale(struct block_list *bl, int src_id, int x, int y, uint16 sk
 #endif
 }
 
+void clif_stylist_vector_init(void)
+{
+	int i;
+	for (i = 0; i < MAX_STYLIST_TYPE; i++) {
+		VECTOR_INIT(stylist_data[i]);
+	}
+}
+
+void clif_stylist_vector_clear(void)
+{
+	int i;
+	for (i = 0; i < MAX_STYLIST_TYPE; i++) {
+		VECTOR_CLEAR(stylist_data[i]);
+	}
+}
+
+bool clif_stylist_read_db_libconfig(void)
+{
+	struct config_t stylist_conf;
+	struct config_setting_t *stylist = NULL, *it = NULL;
+	const char *config_filename = "db/stylist_db.conf"; // FIXME hardcoded name
+	int i = 0;
+
+	if (!libconfig->load_file(&stylist_conf, config_filename))
+		return false;
+
+	if ((stylist = libconfig->setting_get_member(stylist_conf.root, "stylist_db")) == NULL) {
+		ShowError("can't read %s\n", config_filename);
+		return false;
+	}
+
+	clif->stylist_vector_clear();
+
+	while ((it = libconfig->setting_get_elem(stylist, i++))) {
+		clif->stylist_read_db_libconfig_sub(it, i - 1, config_filename);
+	}
+
+	libconfig->destroy(&stylist_conf);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", i, config_filename);
+	return true;
+}
+
+bool clif_stylist_read_db_libconfig_sub(struct config_setting_t *it, int idx, const char *source)
+{
+	struct stylist_data_entry entry = { 0 };
+	int i32 = 0, type = 0;
+	int64 i64 = 0;
+
+	nullpo_ret(it);
+	nullpo_ret(source);
+
+	if (!itemdb->lookup_const(it, "Type", &type) || type > MAX_STYLIST_TYPE || type <= 0) {
```

Comment:
should be ``type > MAX_STYLIST_TYPE`` because if type == MAX_STYLIST_TYPE this is invalid index.